src/main/java/com/minecolonies/api/tileentities/storageblocks/IStorageBlockInterface.java

```diff
@@ -0,0 +1,117 @@
+package com.minecolonies.api.tileentities.storageblocks;
+
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.crafting.ItemStorage;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import java.util.Map;
+import java.util.function.Predicate;
+
+public interface IStorageBlockInterface
+{
+    /**
+     * Whether the storageblock should be included in building containers automatically.
+     *
+     * @param building The building that is being constructed.
+     * @return Whether the storageblock should be included in building containers automatically.
+     */
+    boolean shouldAutomaticallyAdd(final IBuilding building);
+
+    /**
+     * Sets whether the block is part of a warehouse.
+     *
+     * @param inWarehouse Whether it's in a warehouse
+     */
+    void setInWarehouse(final boolean inWarehouse);
+
+    /**
+     * Gets the current upgrade level of the storageblock
+     *
+     * @return The current level
+     */
+    int getUpgradeLevel();
+
+    /**
+     * Upgrades the size of the storage, if applicable.
+     */
+    void increaseUpgradeLevel();
```

Comment:
Same as above

Reply:
Hmm, I'm not convinced on this one. The upgrades will be very storage block dependent. For example, does an upgrade for Iron Chests mean we add new rows to the chests, like we currently do with Racks? Or do we not upgrade the chests at all and actually just move from Iron Chests -> Gold Chests -> Diamond Chests (and only allow them at specific building levels). For Storage Drawers, it probably makes the most sense to just base the number of upgrades a drawer can have based on the building level. For AE2, it'd probably something like the size of disks that are allowed or the number of connections to the AE network.

It seems like it will have to be defined by the owner of the addon.

Reply:
What I'd do instead (because this is actually needed on the block side to set an external level), is give this one a default implementation of nothing.
That way the majority of the blocks won't be forced to implement this with nothing inside of it, since it's not something that the majority of blocks will use.

Alternatively, define a separate interface `IStorageBlockUpgradeable` which contains this method, then the logic is opt-in.